src/NServiceBus.Core/obsoletes.cs

```diff
@@ -686,7 +686,7 @@ public static string GetMessageHeader(this IBus bus, object msg, string key)
         [ObsoleteEx(
             RemoveInVersion = "7.0",
             TreatAsErrorFromVersion = "6.0",
-            Message = "Headers can be set using the ``.SetHeader` method on the context object passed into the behavior or mutator")]
+            Message = @"Use the overload of the Send, Publish or Reply method that accepts an options parameter. Call options.SetHeader(""MyHeader"",""MyValue"") instead.")]
```

Comment:
@timbussmann - What does `The member currently throws a NotImplementedException.` mean? Which member throws an Not implemented exception?